User/ekarabulut/ocp lock update (#994)
chipsalliance/caliptra-rtl: docs/CaliptraIntegrationSpecification.md

|  strap_ss_strap_generic_2                                 | 32  | Input Strap | Synchronous to clk | Used in Subsystem mode only. In Passive mode, integrators shall tie this input to 0.|
|  strap_ss_strap_generic_3                                 | 32  | Input Strap | Synchronous to clk | Used in Subsystem mode only. In Passive mode, integrators shall tie this input to 0.|

> |  ss_debug_intent                                          | 1   | Input | Synchronous to clk | Sample on cold reset. Used in Subsystem mode only. Indicates that the SoC is in debug mode and a user intends to request unlock of debug mode through the TAP mailbox. In Passive mode, integrators shall tie this input to 0. |

Reviewer: ```suggestion
|  ss_debug_intent                                          | 1   | Input | Synchronous to clk | Sampled on cold reset. Used in Subsystem mode only. Indicates that the SoC is in debug mode and a user intends to request unlock of debug mode through the TAP mailbox. In Passive mode, integrators shall tie this input to 0. |
```

> |  ss_debug_intent                                          | 1   | Input | Synchronous to clk | Sample on cold reset. Used in Subsystem mode only. Indicates that the SoC enables OCP LOCK features of Caliptra |

Reviewer: ```suggestion
|  ss_ocp_lock_en                                          | 1   | Input | Synchronous to clk | Sample on cold reset. Used in Subsystem mode only. Indicates that the SoC enables OCP LOCK features of Caliptra |
```

> |  ss_ocp_lock_en                                           | 1   | Input | Synchronous to clk | Sample on cold reset. Used in Subsystem mode only. Indicates that the SoC enables OCP LOCK features of Caliptra. Must be tied to a constant value. For example, driving this input from a programmable register or from a package pin is not permitted. |

Reviewer 2: "Sample" -> "Sampled"

Reviewer: ```suggestion
|  ss_ocp_lock_en                                           | 1   | Input | Synchronous to clk | Sampled on cold reset. Used in Subsystem mode only. Indicates that the SoC enables OCP LOCK features of Caliptra. Must be tied to a constant value. For example, driving this input from a programmable register or from a package pin is not permitted. |
```

| ecc_montgomerymultiplier  | Netlist for always_ff block does not contain flip flop                                    | 274, 326 |Output width is smaller than internal signals, synthesis optimizes away the extra internal flops with no loads|
| Multiple modules          | Signed to unsigned conversion occurs                                                      |          ||

> ## OCP LOCK Caliptra ROM & Firmware Guidance

Reviewer: ```suggestion
## OCP LOCK Caliptra ROM & Firmware Requirements
```

- **DOE error status (KV write failure):**  
  Introduce a new status bit; ROM must check it after any DOE flow to ensure the KV write succeeded.
- **HEK DOE flow**:  
  Must be locked after execution (like other DOE flows). ROM must run the HEK DOE flow even in non-LOCK contexts to lock it. Reset only by hard reset.
- **Derivations & locking**:

>   - Derive the MDK to **KV16**, then **lock writes**.

Reviewer: ```suggestion
  - Derive the MDK to **KV16**, then set the **lock_wr** control bit of KV16.
```

>   - If OCP LOCK is enabled, derive **HEK** to **KV22**, then **lock writes**.

Reviewer: ```suggestion
  - If OCP LOCK is enabled, derive **HEK** to **KV22**, then set the **lock_wr** control bit of KV23.
```

>     - Perform this **before** setting `OCP_LOCK_IN_PROGRESS` since HEK derivation depends on the **HEK seed (KV22)** and **standard CDI derivatives** (from standard KV slots).

Reviewer: ```suggestion
    - Perform this **before** setting `OCP_LOCK_IN_PROGRESS` since HEK derivation depends on the **HEK seed** and **standard CDI derivatives** (from standard KV slots).
```

- **Setting lock state:**  

>   If OCP LOCK is enabled, ROM **MUST** set `OCP_LOCK_IN_PROGRESS` **after any type of reset**.

Reviewer: ```suggestion
  If OCP LOCK is enabled, ROM **MUST** set `OCP_LOCK_IN_PROGRESS` before transitioning to the First Mutable Code (FMC) image.
```

Reviewer: Because OCP_LOCK_IN_PROGRESS is now on cptra_pwrgood

- **KV23 usage constraints:**  

>   Never attempt to write anything to **KV23** except **MEK** (produced by AES).

Reviewer: ```suggestion
  Never attempt to write anything to **KV23** except **MEK** (produced by AES).
```

>   - DOE will flag an error and **stall** if a write to **KV23** is attempted while OCP LOCK is in progress.

Reviewer: ```suggestion
  - DOE will flag an error and **stall** if a write to **KV23** is attempted while OCP LOCK is enabled.
```

- **DMA programming with KV:**  

>   Follow the rules for **size** and **destination address**.

Reviewer: ```suggestion
  The programmed byte count must match the value on the input strap_ss_key_release_key_size strap.  The programmed destination address must match the value on the input strap_ss_key_release_base_addr. Violation of either of these rules will trigger a command decode error in the DMA.
```

- **KV filtering rules:**  
  Standard ↔ LOCK **crossovers are blocked**.
- **AES API updates:**  
  Adjust for KV interactions, `dataout` behavior, and the **`OUTPUT_LOST`** use-case.
- **Strap validation:**  
  Read and confirm valid values for:

>   - `KEY_RELEASE_BASE_ADDR`

Reviewer: ```suggestion
  - [SS_KEY_RELEASE_BASE_ADDR](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.soc_ifc_reg.SS_KEY_RELEASE_SIZE)
```

>   - `KEY_RELEASE_KEY_SIZE`

Reviewer: ```suggestion
  - [KEY_RELEASE_KEY_SIZE](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.soc_ifc_reg.SS_KEY_RELEASE_SIZE)
```
